connector/src/main/java/org/geysermc/connector/entity/living/merchant/VillagerEntity.java

```diff
@@ -97,58 +97,49 @@ public void updateBedrockMetadata(EntityMetadata entityMetadata, GeyserSession s
         }
         super.updateBedrockMetadata(entityMetadata, session);
     }
-    
+
     @Override
     public void moveRelative(GeyserSession session, double relX, double relY, double relZ, Vector3f rotation, boolean isOnGround) {
-        if (!metadata.getFlags().getFlag(EntityFlag.SLEEPING)) {
+        if (!metadata.getFlags().getFlag(EntityFlag.SLEEPING) || metadata.getPos(EntityData.BED_POSITION, null) == null) {
```

Comment:
Instead of running `metadata.getPos(EntityData.BED_POSITION, null)` twice: declare `bedPosition` as a variable above and then replace this with `(bedPosition = metadata.getPos(EntityData.BED_POSITION, null)) == null` so it can be reused if helpful.

Reply:
We could also check getFlag(SLEEPING) before calling getPos

Maybe once I get back to my computer later today